python_modules/dagster/dagster_tests/definitions_tests/auto_materialize_tests/asset_condition_tests/test_backfills_requested.py

```diff
@@ -0,0 +1,197 @@
+from datetime import datetime, timezone
+
+from dagster import AutomationCondition
+from dagster._core.definitions.asset_key import AssetKey
+from dagster._core.definitions.time_window_partitions import TimeWindow
+
+from dagster_tests.definitions_tests.auto_materialize_tests.base_scenario import run_request
+
+from ..scenario_specs import (
+    daily_partitions_def,
+    hourly_partitions_def,
+    two_assets_depend_on_one,
+    two_assets_in_sequence,
+    two_disconnected_graphs,
+    two_partitions_def,
+)
+from .asset_condition_scenario import AutomationConditionScenarioState
+
+
+def test_simple_eager_conditions_with_backfills() -> None:
+    state = (
+        AutomationConditionScenarioState(
+            two_assets_in_sequence,
+            automation_condition=AutomationCondition.eager(),
+            ensure_empty_result=False,
+            request_backfills=True,
+        )
+        .with_asset_properties(partitions_def=hourly_partitions_def)
+        .with_current_time("2020-02-02T01:05:00")
+    )
+
+    # parent hasn't updated yet
+    new_run_requests, _, _ = state.evaluate_daemon_tick(["B"])
+    assert len(new_run_requests) == 0
+
+    # historical parent updated, doesn't matter
+    state = state.with_runs(run_request("A", "2019-07-05-00:00"))
+    new_run_requests, _, _ = state.evaluate_daemon_tick(["B"])
+    assert len(new_run_requests) == 0
+
+    # latest parent updated, now can execute
+    state = state.with_runs(run_request("A", "2020-02-02-00:00"))
+    new_run_requests, _, _ = state.evaluate_daemon_tick(["B"])
+    assert len(new_run_requests) == 1
+    assert new_run_requests[0].requires_backfill_daemon()
+    state = state.with_runs(
+        *(
+            run_request(ak, pk)
+            for ak, pk in new_run_requests[0].asset_graph_subset.iterate_asset_partitions()
+        )
+    )
+
+    # now B has been materialized, so don't execute again
+    new_run_requests, _, _ = state.evaluate_daemon_tick(["B"])
+    assert len(new_run_requests) == 0
+
+    # new partition comes into being, parent hasn't been materialized yet
+    state = state.with_current_time_advanced(hours=1)
+    new_run_requests, _, _ = state.evaluate_daemon_tick(["B"])
+    assert len(new_run_requests) == 0
+
+    # parent gets materialized, B requested
+    state = state.with_runs(run_request("A", "2020-02-02-01:00"))
+    new_run_requests, _, _ = state.evaluate_daemon_tick(["B"])
+    assert len(new_run_requests) == 1
+    # but it fails
+    state = state.with_failed_run_for_asset("B", "2020-02-02-01:00")
+
+    # B does not get immediately requested again
+    new_run_requests, _, _ = state.evaluate_daemon_tick(["B"])
```

Comment:
@OwenKephart something i'm confused about is that this test is failing. afaik nothing changed with how the conditions are evaluated in this PR so i'm not sure why B is getting requested again when this test https://github.com/dagster-io/dagster/blob/40da65bf6ea9191024e0b8bf454474a9519a34c5/python_modules/dagster/dagster_tests/definitions_tests/auto_materialize_tests/asset_condition_tests/test_eager_condition.py#L96-L98 claims it shouldn't get requested again